### PR TITLE
Push BuildScopes and VerifyScopes into ICredentialBuilder

### PR DESCRIPTION
**Pull request by joelverhagen**
This allows more code sharing for other places that create credentials (such as a new endpoint to create a short-lived API key from an OIDC token).

Also, add a method to `IAuthenticationService` which is on the concrete implementation (`AuthenticationService`) to make mocking easier.

Progress on https://github.com/NuGet/NuGetGallery/issues/10212.

### CODE REVIEW

File: src/NuGetGallery.Services/Authentication/IAuthenticationService.cs [hunk 1]
```diff
@@ -43,5 +43,12 @@ public interface IAuthenticationService
         /// </summary>
         /// <returns>Returns whether the API key credential is active or not</returns>
         bool IsActiveApiKeyCredential(Credential credential);
+
+        /// <summary>
+        /// Adds a new credential to the user. This method saves changes in the entity context.
+        /// </summary>
+        /// <param name="user">The user who owns the credental.</param>
+        /// <param name="credential">The credential to be added.</param>
+        Task AddCredential(User user, Credential credential);
```
Comment (zhhyu) [inline src/NuGetGallery.Services/Authentication/IAuthenticationService.cs:52]: nit: this doesn't seem related to this PR.

**Review by zhhyu — APPROVED** [review]

File: src/NuGetGallery.Services/Authentication/IAuthenticationService.cs (hunk 1 above)
Comment (joelverhagen) [reply src/NuGetGallery.Services/Authentication/IAuthenticationService.cs:52]: Sorry, I wanted it in the PR but was too specific in my description. This method is on the concrete implementation but not on the interface, making mocking harder.

**Review by zhhyu — APPROVED** [review]
